GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayClient.kt

```diff
@@ -0,0 +1,636 @@
+    fun getTokenizationParameters(
+        callback: GooglePayGetTokenizationParametersCallback
+    ) {
+        braintreeClient.getConfiguration { configuration: Configuration?, e: Exception? ->
+            if (configuration == null && e != null) {
+                callback.onTokenizationParametersResult(GooglePayTokenizationParameters.Failure(e))
+                return@getConfiguration
+            }
+            callback.onTokenizationParametersResult(
+                GooglePayTokenizationParameters.Success(
+                    getTokenizationParameters(configuration, braintreeClient.authorization),
+                    getAllowedCardNetworks(configuration)
+                )
+            )
+        }
+    }
+
+    /**
+     * Start the Google Pay payment flow. This will return [GooglePayPaymentAuthRequestParams] that are
+     * used to present Google Pay payment sheet in
+     * [GooglePayLauncher.launch]
+     *
+     * @param request  The [GooglePayRequest] containing options for the transaction.
+     * @param callback [GooglePayPaymentAuthRequestCallback]
+     */
+    @SuppressWarnings("LongMethod", "MaximumLineLength")
+    fun createPaymentAuthRequest(
+        request: GooglePayRequest,
+        callback: GooglePayPaymentAuthRequestCallback
+    ) {
+        braintreeClient.sendAnalyticsEvent(GooglePayAnalytics.PAYMENT_REQUEST_STARTED)
+
+        if (!validateManifest()) {
+            callbackPaymentRequestFailure(
+                GooglePayPaymentAuthRequest.Failure(
+                    BraintreeException(
+                        "GooglePayActivity was not found in the Android " +
+                                "manifest, or did not have a theme of R.style.bt_transparent_activity"
+                    )
+                ), callback
+            )
+            return
+        }
+
+        if (request == null) {
+            callbackPaymentRequestFailure(
+                GooglePayPaymentAuthRequest.Failure(
+                    BraintreeException(
+                        "Cannot pass null GooglePayRequest to requestPayment"
+                    )
+                ), callback
+            )
+            return
+        }
+
+        if (request.transactionInfo == null) {
+            callbackPaymentRequestFailure(
+                GooglePayPaymentAuthRequest.Failure(
+                    BraintreeException(
+                        "Cannot pass null TransactionInfo to requestPayment"
+                    )
+                ), callback
+            )
+            return
+        }
+
+        braintreeClient.getConfiguration { configuration: Configuration?, configError: Exception? ->
+            if (configuration == null && configError != null) {
+                callbackPaymentRequestFailure(
+                    GooglePayPaymentAuthRequest.Failure(configError),
+                    callback
+                )
+                return@getConfiguration
+            }
+            if (!configuration!!.isGooglePayEnabled) {
```

Comment:
I overlooked that the configuration variable is possible to be null here because when configuration is null && configError is null. I will probably have to re-think this block.